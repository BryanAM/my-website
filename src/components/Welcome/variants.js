export const parent = {
  fadeIn: {
    transition: { staggerChildren: 0.1, delayChildren: 0.1}
  }
}

export const variant = {
  fadeIn: {
    opacity: 1,
    transition: {duration: 2}
  }    
}
