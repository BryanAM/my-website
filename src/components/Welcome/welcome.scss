@import '../../resources/styles/colors.scss';
@import '../../resources/styles/text.scss';
@import '../../resources/styles/screen.scss';

#welcome-section {
  min-height: 100vh;
}

.language-picker {
  display: inline-block;
  border: solid 1px $accent-pink;
  border-radius: 0.5rem;
  padding: 0.5rem 1rem;
  margin: 0.25rem;
  min-width: 10rem;
  height: auto;
  text-decoration: none;
  background: $nav-blue;
  color: #ffffff;
  font-family: sans-serif;
  font-size: $medium;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  transition: background .5s ease-in-out;
  &:hover,
  &focus {
    background: rgba(230, 103, 167, 0.3);
  }
}
.me-long {
  color: $white;
  font-size: $medium;
  color: $standard-text;
}

.me-name {
  color: $light-text;
  font-size: $large;
  margin-top: 0;
  margin-bottom: 0;
  font-weight: $x-bold;
  @media screen and (min-width: $break-large) {
    font-size: $xx-large;
  }
}

.me-short {
  color: $white;
  font-size: $large;
  margin-top: 0;
  margin-bottom: 0;
  font-weight: $x-bold;
  color: $standard-text;
  @media screen and (min-width: $break-large) {
    font-size: $x-large;
  }
}

.pick-lang-desc {
  color: $accent-pink;
  font-size: $medium;
  margin-top: 0;
  @media screen and (min-width: $break-large) {
    font-size: $large;
  }
}

.section-header {
  margin-top: 2rem;
}

