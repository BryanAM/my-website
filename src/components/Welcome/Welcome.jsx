import React from 'react';
import { useTranslation } from 'react-i18next';
import i18n from '../../i18n/index.js';
import { motion } from "framer-motion";
import { parent, variant } from './variants.js';
import './welcome.scss';

const Welcome = () => {
  const [t] = useTranslation();
  const handleOnClick = () => {
    let switchLang = i18n.language === 'ja' ? 'en' : 'ja';
    i18n.changeLanguage(switchLang, (err, t) => {
      if (err) return console.log('something went wrong loading', err);
    });
  }
  return(
    <motion.section  animate='fadeIn' variants={parent} id='welcome-section' className='section'>
      <motion.h1 initial={{ opacity: 0 }} variants={variant} className='section-header'>{t('welcome.greeting')}</motion.h1>
      <motion.p initial={{ opacity: 0 }} variants={variant} className='me-name'>{t('welcome.name')}</motion.p>
      <motion.p initial={{ opacity: 0 }} variants={variant} className='me-short'>{t('welcome.about')}</motion.p>
      <motion.p initial={{ opacity: 0 }} variants={variant} className='me-long'>{t('welcome.detail')}</motion.p>
      <motion.p initial={{ opacity: 0 }} variants={variant} className='pick-lang-desc'>{t('welcome.switchDesc')}</motion.p>
      <motion.button initial={{ opacity: 0 }} variants={variant} className='language-picker' onClick={handleOnClick}>
        {t('welcome.switch')}
      </motion.button>
    </motion.section>
  );
};

export default Welcome;