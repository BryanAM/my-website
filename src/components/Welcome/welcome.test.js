import React from 'react';
import { shallow } from 'enzyme';

import Welcome from './Welcome.jsx';

describe('Welcome', () => {
  it('should render correctly', () => {
    const component = shallow(<Welcome />);
    expect(component).toMatchSnapshot();
  });
});