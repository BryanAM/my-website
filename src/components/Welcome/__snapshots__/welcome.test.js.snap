// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should render correctly 1`] = `
<ForwardRef(MotionComponent)
  animate="fadeIn"
  className="section"
  id="welcome-section"
  variants={
    Object {
      "fadeIn": Object {
        "transition": Object {
          "delayChildren": 0.1,
          "staggerChildren": 0.1,
        },
      },
    }
  }
>
  <ForwardRef(MotionComponent)
    className="section-header"
    initial={
      Object {
        "opacity": 0,
      }
    }
    variants={
      Object {
        "fadeIn": Object {
          "opacity": 1,
          "transition": Object {
            "duration": 2,
          },
        },
      }
    }
  >
    welcome.greeting
  </ForwardRef(MotionComponent)>
  <ForwardRef(MotionComponent)
    className="me-name"
    initial={
      Object {
        "opacity": 0,
      }
    }
    variants={
      Object {
        "fadeIn": Object {
          "opacity": 1,
          "transition": Object {
            "duration": 2,
          },
        },
      }
    }
  >
    welcome.name
  </ForwardRef(MotionComponent)>
  <ForwardRef(MotionComponent)
    className="me-short"
    initial={
      Object {
        "opacity": 0,
      }
    }
    variants={
      Object {
        "fadeIn": Object {
          "opacity": 1,
          "transition": Object {
            "duration": 2,
          },
        },
      }
    }
  >
    welcome.about
  </ForwardRef(MotionComponent)>
  <ForwardRef(MotionComponent)
    className="me-long"
    initial={
      Object {
        "opacity": 0,
      }
    }
    variants={
      Object {
        "fadeIn": Object {
          "opacity": 1,
          "transition": Object {
            "duration": 2,
          },
        },
      }
    }
  >
    welcome.detail
  </ForwardRef(MotionComponent)>
  <ForwardRef(MotionComponent)
    className="pick-lang-desc"
    initial={
      Object {
        "opacity": 0,
      }
    }
    variants={
      Object {
        "fadeIn": Object {
          "opacity": 1,
          "transition": Object {
            "duration": 2,
          },
        },
      }
    }
  >
    welcome.switchDesc
  </ForwardRef(MotionComponent)>
  <ForwardRef(MotionComponent)
    className="language-picker"
    initial={
      Object {
        "opacity": 0,
      }
    }
    onClick={[Function]}
    variants={
      Object {
        "fadeIn": Object {
          "opacity": 1,
          "transition": Object {
            "duration": 2,
          },
        },
      }
    }
  >
    welcome.switch
  </ForwardRef(MotionComponent)>
</ForwardRef(MotionComponent)>
`;
